odcockpit/src/plugins/environmentviewer/EnvironmentViewerGLWidget.cpp

```diff
@@ -391,6 +469,15 @@ namespace cockpit {
             }
 
             void EnvironmentViewerGLWidget::nextContainer(Container &c) {
+                
+                if(c.getDataType()==odcore::data::SharedPointCloud::ID()){
+                    velodyneFrame=c.getData<SharedPointCloud>();//Get shared point cloud
+                    if (!m_hasAttachedToSharedImageMemory) {
```

Comment:
I tried this, but then part of the point cloud is missing in the replay. So it means that this statement has to be outside the if block.